android/src/main/java/com/urbanairship/reactnative/EventEmitter.java

```diff
@@ -110,6 +114,48 @@ public void onReactContextInitialized(ReactContext reactContext) {
         }
     }
 
+    /**
+     * Called when a new listener is added for a specified event name.
+     *
+     * @param eventName The event name.
+     */
+    void addAndroidListener(ReactContext reactContext, String eventName) {
+        synchronized (knownListeners) {
+            List<Event> pending = new ArrayList<>();
+            pending.addAll(pendingEvents);
+
+            for (Event event : pendingEvents) {
+                if (event.equals(eventName)) {
+                    sendEvent(reactContext, event);
+                    pending.remove(event);
+                }
+            }
+
+            pendingEvents.clear();
+            pendingEvents.addAll(pending);
+
+            listenerCount++;
+            knownListeners.add(eventName);
+        }
+    }
+
+    /**
+     * Called when listeners are removed.
+     *
+     * @param count The count of listeners.
+     */
+    void removeAndroidListeners(ReactContext reactContext, int count) {
+        synchronized (knownListeners) {
+            long currentCount = listenerCount;
+            listenerCount = max(0, currentCount - count);
+
+            if (listenerCount == 0) {
+                setEnabled(reactContext, false);
```

Comment:
do we still need the enabled flag?

Reply:
The code in setEnabled could cause an Exception. I dont think its needed anymore so you might be able to just delete it.